### PR TITLE
Rekey status command, ShowPendingRekeyStatus RPC

### PR DESCRIPTION
**Pull request by patrickxb**
This adds `rekey status` and `ShowPendingRekeyStatus`.  There is a stub for `rekey trigger` which is a debugging command to trigger a rekey status window.

### CODE REVIEW

**Comment by cjb** [comment]
CI failures on this one

**Comment by patrickxb** [comment]
Log looked like OS X instance didn't spin up?

**Comment by cjb** [comment]
Hm? This is the log:

https://ci.keybase.io/job/client-osx/215/label=osx/console

```
21:14:51 service.......# github.com/keybase/client/go/service
21:15:07 ./crypto_test.go:25: cannot use fakeUIRouter literal (type fakeUIRouter) as type libkb.UIRouter in argument to tc.G.SetUIRouter:
21:15:07 	fakeUIRouter does not implement libkb.UIRouter (missing GetRekeyUINoSessionID method)
```

**Comment by patrickxb** [comment]
Well that's new. First one was differ t. 

I'll fix. There's no need to comment that CI is failing, right? We all see the green dots? Or is something new with Jenkins that you need to write comments manually?

BTW, this was in case nojima needed the branch.

**Comment by cjb** [comment]
> There's no need to comment that CI is failing, right?

Yeah, just happened to be looking at it to verify CI was working -- it does show up in the status, no good reason for my comment.  :)

**Comment by patrickxb** [comment]
@keybase/react-hackers I think I have flow issues in this branch.  Can anyone help?

**Comment by patrickxb** [comment]
This is ready.  @chrisnojima `rekey trigger` should work as we discussed.  @maxtaco PTAL.

File: go/service/ui_router.go [hunk 1]
```diff
@@ -155,3 +155,20 @@ func (u *UIRouter) GetRekeyUI() (keybase1.RekeyUIInterface, int, error) {
 	}
 	return ret, sessionID, nil
 }
+
+func (u *UIRouter) GetRekeyUINoSessionID() (keybase1.RekeyUIInterface, error) {
```
Comment (maxtaco) [inline go/service/ui_router.go]: Why is this needed?

**Comment by maxtaco** [comment]
Had a question about the `NoSessionID` getter

File: go/service/ui_router.go (hunk 1 above)
Comment (patrickxb) [inline go/service/ui_router.go]: The other one (GetRekeyUI) requests a session id from electron.  This is for the one-off interactions where CLI would like to specify 0 for session id so that electron knows that it is a one-off UI interaction.

Comment (patrickxb) [inline go/service/ui_router.go]: But @chrisnojima if you'd prefer service to ask for a session id first, i'm fine with that.

Comment (chrisnojima) [inline go/service/ui_router.go]: our typical logic is if there's no sessionid, it was spawned by the daemon. if it has a sessionid its cause electron spawned it and it using it to bookkeep.

Comment (patrickxb) [inline go/service/ui_router.go]: Right, that's why I did it this way since the user is initiating these calls via CLI.

The gregor initiated calls, which result in a refresh loop, ask electron for a session id so that electron can bookkeep the window.

Comment (maxtaco) [inline go/service/ui_router.go]: ok, so this is different from the normal case of CLI-versions of commands where the UID need not be delegated?

File: go/client/cmd_rekey.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	"github.com/keybase/client/go/libcmdline"
+	"github.com/keybase/client/go/libkb"
+)
+
+// These are the production rekey commands
+func NewCmdRekey(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	return cli.Command{
+		Name:         "rekey",
+		Usage:        "Rekey status and actions",
+		ArgumentHelp: "[arguments...]",
```
Comment (maxtaco) [inline go/client/cmd_rekey.go]: better help?

File: go/client/cmd_rekey_devel.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	"github.com/keybase/client/go/libcmdline"
+	"github.com/keybase/client/go/libkb"
+)
+
+// These are the production rekey commands
+func NewCmdRekey(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	return cli.Command{
+		Name:         "rekey",
+		Usage:        "Rekey status and actions",
+		ArgumentHelp: "[arguments...]",
```
Comment (maxtaco) [inline go/client/cmd_rekey_devel.go]: better help?

File: go/service/ui_router.go (hunk 1 above)
Comment (patrickxb) [inline go/service/ui_router.go]: Correct.  In both uses, it is only for interaction with a delegated ui.

**Comment by maxtaco** [comment]
👍 after fixing up the documentation..

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Changed: **Folders_Ignore_Confirmation-Normal_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1465915820737-daf08138bf81-89184d3a2395/Folders_Ignore_Confirmation-Normal_Private-daf08138bf81.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1465915820737-daf08138bf81-89184d3a2395/Folders_Ignore_Confirmation-Normal_Private-89184d3a2395.png)  
   Folders_Ignore_Confirmation-Normal_Private comparison

**Comment by patrickxb** [comment]
retest this please
